app/controllers/videos_controller.rb

```diff
@@ -0,0 +1,18 @@
+class  VideosController < ApplicationController
+  #before_action :require_user
+  before_filter :require_user
+  
+  def index
+    @videos = Video.all
```

Comment:
There is no need to call for `Video.all`, you should use association `@categories.videos` - as each video should have a category. And that should be your access to all videos.